Source/santasyncservice/syncv1.proto

```diff
@@ -0,0 +1,225 @@
+  // retrieve new configuration values from the server.
+  rpc Preflight(PreflightRequest) returns (PreflightResponse) {}
+
+  // EventUpload sends details about blocked (or would-be-blocked) executions to
+  // the server.
+  rpc EventUpload(EventUploadRequest) returns (EventUploadResponse) {}
+
+  // RuleDownload retrieves new rules from the server.
+  rpc RuleDownload(RuleDownloadRequest) returns (RuleDownloadResponse) {}
+
+  // Postflight sends status data back to the server after applying any new
+  // configuration received in the previous stages.
+  rpc Postflight(PostflightRequest) returns (PostflightResponse) {}
+}
+
+// ClientMode represents the operating mode for an agent.
+enum ClientMode {
+  MONITOR = 0;
+  LOCKDOWN = 1;
+}
+
+message PreflightRequest {
+  string serial_number = 1 [json_name="serial_num"];
+  string hostname = 2;
+  string os_version = 3;
+  string os_build = 4;
+  string model_identifier = 5;
+  string santa_version = 6;
+  string primary_user = 7;
+  string push_notification_token = 8;
+
+  ClientMode client_mode = 9;
+  bool request_clean_sync = 10;
+
+  uint32 binary_rule_count = 11;
+  uint32 certificate_rule_count = 12;
+  uint32 compiler_rule_count = 13;
+  uint32 transitive_rule_count = 14;
+  uint32 teamid_rule_count = 15;
+  uint32 signingid_rule_count = 16;
+  uint32 cdhash_rule_count = 17;
+}
+
+message PreflightResponse {
+  ClientMode client_mode = 1;
+  // Possible values are "normal" (default if unspecified), "clean", or "clean_all".
+  // "normal" is a standard progressive sync
+  // "clean" deletes all previously received rules before applying the newly received rules.
+  // "clean_all" deletes all rules, including transitive rules, before applying the newly received rules.
+  // This should be an enum but we won't match lowercase values if specified as an enum unless the
+  // enum values are also lowercase (which is odd), and this is the behavior from the pre-proto protocol.
+  string sync_type = 2;
+
+  // Deprecated but still supported key that acts like sync_type was set to "clean" unless
+  // the client had requested a clean sync, in which case it acts like "clean_all"
+  uint32 batch_size = 3;
+
+  bool enable_bundles = 4;
+  bool enable_transitive_rules = 5;
+  bool enable_all_event_upload = 6;
+  bool disable_unknown_event_upload = 7;
+
+  uint64 full_sync_interval = 8;
+  uint64 push_notification_full_sync_interval = 9;
+  uint64 push_notification_global_rule_sync_interval = 10;
+
+  optional string allowed_path_regex = 11;
+  optional string blocked_path_regex = 12;
+
+  optional bool block_usb_mount = 13;
+  repeated string remount_usb_mode = 14;
+
+  optional string override_file_access_action = 15;
+
+  // These fields are deprecated forms of other fields and exist here solely for backwards compatibility
+  optional bool deprecated_enabled_transitive_whitelisting = 80 [json_name="enabled_transitive_whitelisting"];
```

Comment:
What about using the `[deprecated=true]` tag instead of the `deprecated_` prefix in the name?